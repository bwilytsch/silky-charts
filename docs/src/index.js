import React from 'react';
import { render } from 'react-dom';
import { Router, Redirect } from '@reach/router';
import Layout from './containers/Layout';
import Header from './containers/Header';
import Navigation from './containers/Navigation';
import Content from './containers/Content';
import Introduction from './components/Introduction';
import BarChart from './components/BarChart';
import pages from './pages.json';
import 'normalize.css';
import './styles.css';

const App = () => {
  const baseURL = '/silky-charts';
  const pathname = window.location.pathname;
  const currentPath = pathname.includes(baseURL)
    ? pathname.replace(baseURL, '')
    : '/';
  const isValidPathname =
    currentPath && pages.map(({ path }) => path).includes(currentPath);

  console.log(currentPath);

  const contentComponents = {
    Introduction,
    BarChart,
  };

  return (
    <Layout>
      <Header />
      <Navigation
        items={pages}
        baseURL={baseURL}
        initial={isValidPathname ? currentPath : '/'}
      />
      <Router primary={false}>
        {currentPath === '/' && (
          <Redirect from={currentPath} to={baseURL} noThrow />
        )}
        <Content path={baseURL}>
          {pages.map(({ component, path }, idx) => {
            const Component = contentComponents[component];
            return <Component key={idx} path={path} default={path === '/'} />;
          })}
        </Content>
      </Router>
    </Layout>
  );
};

render(<App />, document.getElementById('app'));
