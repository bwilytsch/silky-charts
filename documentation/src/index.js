import React from 'react';
import { render } from 'react-dom';
import { Router, Redirect } from '@reach/router';
import Layout from './containers/Layout';
import Header from './containers/Header';
import Navigation from './containers/Navigation';
import Content from './containers/Content';
import Introduction from './components/Introduction';
import BarChart from './components/BarChart';
import pages from './pages.json';
import 'normalize.css';
import './styles.css';

const App = () => {
  const pathname = window.location.pathname;
  const defaultPath = '/';
  const currentPath =
    pathname !== defaultPath ? pathname.replace(defaultPath, '') : pathname;
  const isInvalidPath = !pages
    .map(({ pathname }) => pathname)
    .includes(currentPath);

  const contentComponents = {
    Introduction,
    BarChart,
  };

  return (
    <Layout>
      <Header />
      <Navigation
        items={pages}
        initialPath={isInvalidPath ? defaultPath : currentPath}
      />
      <Content>
        <Router>
          {isInvalidPath && (
            <Redirect from={currentPath} to={defaultPath} noThrow />
          )}
          {pages.map(({ component, pathname }, idx) => {
            const Component = contentComponents[component];
            return (
              <Component
                key={idx}
                path={pathname}
                default={pathname === defaultPath}
              />
            );
          })}
        </Router>
      </Content>
    </Layout>
  );
};

render(<App />, document.getElementById('app'));
