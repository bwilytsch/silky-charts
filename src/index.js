export { default as Bar } from './Bar';
export { default as BarLine } from './BarLine';
export { default as StackedArea } from './StackedArea';

export { default as toSeries } from './utils/toSeries';
