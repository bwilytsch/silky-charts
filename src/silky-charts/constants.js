// Color pallete
export const pallete = {
  white: 'rgb(255, 255, 255)', // #FFFFFF
  black: 'rgb(33, 33, 33)', // #212121
  grid: 'rgb(220, 220, 220)', // #DCDCDC
  reference: 'rgb(170, 93, 0)', //#AA5D00
  bg: {
    blue: 'rgb(10, 48, 85)', // #0A3055
    gray: 'rgb(46, 52, 59)', // #2E343B
    green: 'rgb(17, 76, 42)', // #114C2A
    orange: 'rgb(102, 56, 15)', // #66380F
    purple: 'rgb(89, 29, 119)', // #591D77
    red: 'rgb(93, 18, 18)', // #5D1212
    yellow: 'rgb(90, 68, 13)', // #5A440D
  },
  fg: {
    blue: 'rgb(15, 72, 128)', // #0F4880
    gray: 'rgb(79, 90, 101)', // #4F5A65
    green: 'rgb(32, 96, 60)', // #20603C
    orange: 'rgb(116, 78, 46)', // #744E2E
    purple: 'rgb(118, 0, 168)', // #7600A8
    red: 'rgb(139, 0, 0)', // #8B0000
    yellow: 'rgb(85, 83, 68)', // #555344
  },
}
