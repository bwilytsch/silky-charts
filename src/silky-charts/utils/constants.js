// Scales
export const SCALE_BAND = 'band';
export const SCALE_LINEAR = 'linear';
export const SCALE_TIME = 'time';

export const TICKS = 5;

// Themes
export const THEME = 'tealBlue';
export const SECONDARY_THEME = 'pink';
