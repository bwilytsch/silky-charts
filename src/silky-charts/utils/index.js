export { default as allDate } from './allDate';
export { default as drawGrid } from './drawGrid';
export { default as extendXPath } from './extendXPath';
export { default as getDivergence } from './getDivergence';
export { default as getXScale } from './getXScale';
export { default as getYScale } from './getYScale';
export { default as rotateXLabels } from './rotateXLabels';
export { default as themes } from './themes';
