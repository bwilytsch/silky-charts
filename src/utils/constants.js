import {
  curveBasis,
  curveBasisClosed,
  curveBasisOpen,
  curveBundle,
  curveCardinal,
  curveCardinalClosed,
  curveCardinalOpen,
  curveCatmullRom,
  curveCatmullRomClosed,
  curveCatmullRomOpen,
  curveLinear,
  curveLinearClosed,
  curveMonotoneX,
  curveMonotoneY,
  curveNatural,
  curveStep,
  curveStepAfter,
  curveStepBefore,
} from 'd3-shape';

// Base values
export const ASPECT_RATIO = '16:9';
export const MARGIN = { top: 40, right: 50, bottom: 50, left: 50 };
export const ROTATION = -50;
export const SIZE = {
  width: 0,
  height: 0,
};
export const X_TICKS = 10;
export const Y_TICKS = 5;
export const TIME_FORMAT = '%a %d';
export const TOOLTIP_DATE_FORMAT = '%b %d, %Y';
export const TOOLTIP_OFFSET = 20;
export const WIDTH = 640;

// Scales
export const SCALE_BAND = 'band';
export const SCALE_LINEAR = 'linear';
export const SCALE_PADDING = 0.15;
export const SCALE_TIME = 'time';

// Themes
export const THEME = 'monteCarlo';
export const SECONDARY_THEME = 'vividCerise';
export const OUTLINE_FILL = 50;
export const OUTLINE_HOVER = 30;
export const OUTLINE_STROKE = -20;

// Line options
export const LINE_STROKE_WIDTH = 2;
export const LINE_TYPE = 'curveLinear';
export const LINE_TYPES = {
  curveBasis,
  curveBasisClosed,
  curveBasisOpen,
  curveBundle,
  curveCardinal,
  curveCardinalClosed,
  curveCardinalOpen,
  curveCatmullRom,
  curveCatmullRomClosed,
  curveCatmullRomOpen,
  curveLinear,
  curveLinearClosed,
  curveMonotoneX,
  curveMonotoneY,
  curveNatural,
  curveStep,
  curveStepAfter,
  curveStepBefore,
};
