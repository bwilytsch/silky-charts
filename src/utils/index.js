export { default as allDate } from './allDate';
export { default as appendStackedValues } from './appendStackedValues';
export { default as buildStack } from './buildStack';
export { default as bySeries } from './bySeries';
export { default as classify } from './classify';
export { debounce } from './debounce';
export { debounceImmediate } from './debounce';
export { default as drawGrid } from './drawGrid';
export { default as extendXPath } from './extendXPath';
export { default as getBaseColor } from './getBaseColor';
export { default as getHoverColor } from './getHoverColor';
export { default as getId } from './getId';
export { default as getLength } from './getLength';
export { default as getLineDataForSeries } from './getLineDataForSeries';
export { default as getMax } from './getMax';
export { default as getStackedMax } from './getStackedMax';
export { default as getSeries } from './getSeries';
export { default as getSize } from './getSize';
export { default as getXScale } from './getXScale';
export { default as getYScale } from './getYScale';
export { default as isValidDate } from './isValidDate';
export { default as palette } from './palette';
export { default as rotateXLabels } from './rotateXLabels';
export { default as setLineType } from './setLineType';
export { default as setupData } from './setupData';
export { default as toSingleForm } from './toSingleForm';
export { default as toStackedForm } from './toStackedForm';
export { default as valueFor } from './valueFor';
