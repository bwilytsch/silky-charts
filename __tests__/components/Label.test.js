import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { Label } from '../../src/components';
import { MARGIN } from '../../src/utils/constants';
import 'jest-styled-components';

const props = {
  height: 10,
  margin: MARGIN,
  width: 10,
};

afterEach(cleanup);

test('Label:X', () => {
  const { container } = render(
    <svg>
      <Label axis="x" {...props} />
    </svg>
  );

  const label = container.querySelector('text');
  expect(label).toMatchSnapshot();
  expect(label.classList.contains('x-axis-label')).toBe(true);
});

test('Label:Y', () => {
  const { container } = render(
    <svg>
      <Label axis="y" {...props} />
    </svg>
  );

  const label = container.querySelector('text');
  expect(label).toMatchSnapshot();
  expect(label.classList.contains('y-axis-label')).toBe(true);
});
