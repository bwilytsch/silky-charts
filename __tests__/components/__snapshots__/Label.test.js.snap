// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label:X 1`] = `
.c0 {
  text-anchor: middle;
}

<text
  class="x-axis-label c0"
  height="10"
  width="10"
  x="5"
  y="30"
/>
`;

exports[`Label:Y 1`] = `
.c0 {
  text-anchor: middle;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

<text
  class="y-axis-label c0"
  height="10"
  width="10"
  x="-5"
  y="-16"
/>
`;
